### PR TITLE
[xcode11] Fix several test and API issues that made apitest, introspection and xammac tests fail on macOS 10.15.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: tests/apitest/apitest.csproj [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <ConsolePause>false</ConsolePause>
-    <EnableCodeSigning>false</EnableCodeSigning>
+    <EnableCodeSigning>true</EnableCodeSigning>
```
Comment (dalexsoto) [inline tests/apitest/apitest.csproj]: Was this an intentional change? Just asking for a friend :P

File: tests/apitest/src/DelegateAndDataSourceTest.cs [hunk 2]
```diff
@@ -132,6 +132,14 @@ bool Skip (Type t)
...
 			case "QTMovie":
 				return TestRuntime.CheckSystemVersion (PlatformName.MacOSX, 10, 14, 4); // Broke in macOS 10.14.4.
+			case "AVCaptureView":
+				// Deallocating a AVCaptureView makes it trigger a permission dialog, which we don't want, so just skip this type.
+				return true;
+			}
+
+			switch (t.Namespace) {
+			case "QTKit":
+				return TestRuntime.CheckSystemVersion (PlatformName.MacOSX, 10, 15); // QTKit is gone in 10.15.
```
Comment (spouliot) [inline tests/apitest/src/DelegateAndDataSourceTest.cs]: 👋

File: tests/introspection/Mac/introspection-mac.csproj [hunk 3]
```diff
@@ -23,13 +23,14 @@
     <DefineConstants>DEBUG;MONOMAC;$(DefineConstants)</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <EnableCodeSigning>false</EnableCodeSigning>
+    <EnableCodeSigning>true</EnableCodeSigning>
```
Comment (dalexsoto) [inline tests/introspection/Mac/introspection-mac.csproj]: same?

File: tests/apitest/src/DelegateAndDataSourceTest.cs (hunk 2 above)
Comment (dalexsoto) [reply tests/apitest/src/DelegateAndDataSourceTest.cs]: Farewell!

File: tests/apitest/apitest.csproj (hunk 1 above)
Comment (rolfbjarne) [reply tests/apitest/apitest.csproj]: Yes, otherwise the permission keys in the Info.plist don't work.

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-6431/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-6431/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Running XM tests on '10.15', Running XM tests on '10.15'' 🔥

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-6431/1//console)
✅ Packages: [xamarin.ios-12.99.0.164.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/package/xamarin.ios-12.99.0.164.pkg) [xamarin.mac-5.99.0.164.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/package/xamarin.mac-5.99.0.164.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/jenkins-results/generator-diff/index.html) (no change)
🔥 [Xamarin.Mac tests on 10.15 failed: Xamarin.Mac tests on macOS 10.15 failed (xammac_tests)](https://jenkins.internalx.com/job/macios/job/PR-6431/1/display/redirect) 🔥
🔥 [Test run failed](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/5dab3372ab7e0302432ec9b449d5d3c5a4de6ac0/1/jenkins-results/tests/index.html) 🔥

# Test results

<details>
<summary>8 tests failed, 226 tests passed.</summary>

## Failed tests

 * Xtro/Mac: BuildFailure
 * introspection/iOS Unified 32-bits - simulator/Debug: Failed
 * introspection/iOS Unified 32-bits - simulator/Debug (iOS 10.3): Failed
 * introspection/iOS Unified 64-bits - simulator/Debug: Failed
 * introspection/tvOS - simulator/Debug: Failed
 * introspection/watchOS 32-bits - simulator/Debug: Failed
 * introspection/iOS Unified 64-bits - simulator/Debug (iOS 10.3): Failed
 * introspection/tvOS - simulator/Debug (tvOS 10.2): Failed
</details>

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-6431/2/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-6431/2/display/redirect) (on internal Jenkins) failed in stage(s) 'Test run, Test run' 🔥

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-6431/2//console)
✅ Packages: [xamarin.ios-12.99.0.167.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/package/xamarin.ios-12.99.0.167.pkg) [xamarin.mac-5.99.0.167.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/package/xamarin.mac-5.99.0.167.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/jenkins-results/generator-diff/index.html) (no change)
🔥 [Test run failed](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/f934cd7bee84851eec13b2c48d1d5781a2f0922c/2/jenkins-results/tests/index.html) 🔥

# Test results

<details>
<summary>2 tests failed, 232 tests passed.</summary>

## Failed tests

 * Xtro/Mac: Failed (Test run failed.)
 * monotouch-test/watchOS 32-bits - simulator/Debug: Crashed
</details>

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-6431/3/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-6431/3//console)
✅ Packages: [xamarin.ios-12.99.0.179.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/package/xamarin.ios-12.99.0.179.pkg) [xamarin.mac-5.99.0.179.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/package/xamarin.mac-5.99.0.179.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/jenkins-results/apicomparison/api-diff.html) (please review changes)
ℹ️ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/jenkins-results/generator-diff/index.html) (please review changes)
✅ [Test run succeeded](http://xamarin-storage/jenkins/xamarin-macios/PR-6431/da0793775ec09678a811438f257c3b34bc67cf07/3/jenkins-results/tests/index.html)

**Pull request merged**